src/components/structures/login/Registration.js

```diff
@@ -209,24 +210,43 @@ module.exports = React.createClass({
                 accessToken: response.access_token
             });
 
-            // Auto-join rooms
-            if (self.state.teamsConfig && self.state.teamsConfig.teams) {
-                for (let i = 0; i < self.state.teamsConfig.teams.length; i++) {
-                    let team = self.state.teamsConfig.teams[i];
-                    if (self.state.formVals.email.endsWith(team.domain)) {
-                        console.log("User successfully registered with team " + team.name);
+            if (
+                self._rtsClient &&
+                self.props.queryParams &&
+                self.props.queryParams.referrer &&
+                self.state.teamSelected
+            ) {
+                // Track referral, get team_token in order to retrieve team config
+                self._rtsClient.trackReferral(
+                    self.props.queryParams.referrer,
+                    response.user_id,
+                    self.state.formVals.email
+                ).then((args) => {
+                    const teamToken = args[1].team_token;
+                    // Store for use /w welcome pages
+                    window.localStorage.setItem('mx_team_token', teamToken);
```

Comment:
Had a chat about what we could do here & concluded that yep, we just need to check with the team server on login to see what team a newly logged in user is in, if any.